File: packages/flutter_reactive_ble/android/src/main/kotlin/com/signify/hue/flutterreactiveble/model/ConnectionState.kt
package com.signify.hue.flutterreactiveble.model

import com.polidea.rxandroidble2.RxBleConnection

@Suppress("Detekt.MagicNumber")
enum class ConnectionState(val code: Int) {
    CONNECTING(0),
    CONNECTED(1),
    DISCONNECTING(2),
    DISCONNECTED(3),
    UNKNOWN(4)
}

fun RxBleConnection.RxBleConnectionState.toConnectionState(): ConnectionState =
    when (this.name) {
        "DISCONNECTED" -> ConnectionState.DISCONNECTED
        "CONNECTING" -> ConnectionState.CONNECTING
        "CONNECTED" -> ConnectionState.CONNECTED
        "DISCONNECTING" -> ConnectionState.DISCONNECTING
        else -> ConnectionState.UNKNOWN
}

File: packages/reactive_ble_mobile/darwin/reactive_ble_mobile/Sources/reactive_ble_mobile/BleData extras/ConnectionState.swift
import enum CoreBluetooth.CBPeripheralState

func encode(_ connectionState: CBPeripheralState) -> Int32 {
    switch connectionState {
    case .disconnected:
        return 3
    case .connecting:
        return 0
    case .connected:
        return 1
    case .disconnecting:
        return 2
    }
}
